From 0d06d4f4879af39f1bba4b17eb6aeb2e5a68832b Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 20 Jul 2018 15:24:20 +0800
Subject: [PATCH 4464/5242] MLK-18990 gpu: imx: dpu: framegen: Don't set
 clk_disp rate for TMDS encoder

commit  68d9b913c1247da05ca947e528e0c0ff8c39f62f from
https://source.codeaurora.org/external/imx/linux-imx.git

When TMDS encoder is used, the encoder would provide framegen display clock
directly via clk_bypass.  So, we don't have to set clk_disp rate.  This
should work with or without pixel combiner(pixel combiner would combine
two framegens' output to drive high pixel rate displays via TMDS encoder
only currently).

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit 4b710fe10f0f6c6a41b1e15b13b04bb355997e5c)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-framegen.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/gpu/imx/dpu/dpu-framegen.c b/drivers/gpu/imx/dpu/dpu-framegen.c
index bb75bf8..fc2db99 100644
--- a/drivers/gpu/imx/dpu/dpu-framegen.c
+++ b/drivers/gpu/imx/dpu/dpu-framegen.c
@@ -292,9 +292,6 @@ void framegen_shdtokgen(struct dpu_framegen *fg)
 	if (devtype->has_disp_sel_clk && encoder_type_has_tmds) {
 		clk_set_parent(fg->clk_disp_sel, fg->clk_bypass);
 
-		clk_get_rate(fg->clk_disp);
-		clk_set_rate(fg->clk_disp, disp_clock_rate);
-
 		fg->use_bypass_clk = true;
 	} else {
 		/* find an even divisor for PLL */
-- 
1.7.9.5

